### ISSUE TITLE
Memory Issues Problem

### ISSUE THREAD

**Issue by chittelalavanya18** [SDWebImage/SDWebImage#2513]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by chittelalavanya18** [comment]
When i load 100 or more images in tableview using SDWebImage getting Message from debugger: Terminated due to memory issue, and after app is crashing.

**Comment by chittelalavanya18** [comment]
Could you plz provide how to handle memory issue when ever I loaded more images in tableview scrolling time app crash

**Comment by chittelalavanya18** [comment]
it's showing very hight in Energy Impact, How to handle energy impact is low

**Comment by chittelalavanya18** [comment]
![screen shot 2018-10-26 at 6 16 15 pm](https://user-images.githubusercontent.com/12976709/47567251-a80d0400-d94b-11e8-8d0f-3fa5e53a49bb.png)

**Comment by chittelalavanya18** [comment]
I am using DecompressImage in sdwebimage , its getting high memory impact.
How to reduce low memory impact.

**Comment by grimmdaniel** [comment]
I have encountered the same issue (it crashes instantly, with "_Message from debugger: Terminated due to memory issue_"), but for me, the problem only occurred when I used in the options this: options: [.progressiveDownload]. Without this, it's working normally.

**Comment by zhongwuzw** [comment]
@grimmdaniel What's the version of `SD` you use? Can you fill some info like below?

Info | Value
-- | --
Platform Name | e.g. ios / macos / tvos / watchos
Platform Version | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
SDWebImage Version | e.g. 4.2.0 / 4.1.0
Integration Method | e.g. carthage / cocoapods / manually
Xcode Version | e.g. Xcode 9 / Xcode 8
Repro rate | e.g. all the time (100%) / sometimes x% / only once
Repro with our demo prj | e.g. does it happen with our demo project?
Demo project link | e.g. link to a demo project that highlights the issue

**Comment by grimmdaniel** [comment]
> @zhongwuzw  of course :)
> 

> Platform Name:	iOS
> Platform Version:	10.3 and above
> SDWebImage Version:	4.0
> Integration Method:	cocoapods
> Xcode Version:	latest Xcode 10.0 (10A255)
> Repro rate:	all the time (100%)
> Repro with our demo prj:	haven't tested with demo project
> Demo project link:	e.g. link to a demo project that highlights the issue

**Comment by zhongwuzw** [comment]
@grimmdaniel 👍 Unfortunately, we may have this issue in `4.x`, we already do the first step to try to fix this issue, please see #2475(other optimize still in discussion, please see #2477 ) , it's already merged into `master`, but still not release currently. If you need progressive feature, I suggest you fork `SD`, integrate it by using `CocoaPods`  directing to your own `Git` url.

```
pod 'SDWebImage', :git => 'https://github.com/UserName/SDWebImage.git'
```

**Comment by purepure** [comment]
In my case:
I polulate image on a collection view and trying to scroll as fast as I can, then Image on the cell will disappear will canceling download progress -> crash after some scrolling with message "Memory pressure"

**Comment by pushkardeshmukh1992** [comment]
Any update on this? Facing similar issue.
